### DISCUSSION TITLE
Cloud Optimized Point Cloud

### DISCUSSION THREAD

**Discussion by wiesehahn** [r-lidar/lidR#520] [General]
Would it be possible and does it make sense to implement the processing (read/write) of Cloud Optimized Point Cloud Data? 

https://copc.io/

**Comment by Jean-Romain** [comment]
Does it make sense? Yes definitively. Do I plan to make it? No.

`lidR` uses `rlas` to read/write LAS/LAZ files. `rlas` is only a wrapper for R around `LASlib` by Martin Isenburg. I already discussed about `copc` with Howard Butler. If Martin Isenburg was still alive he may have added the support of copc in `LASlib`. On my side it is beyond my skills to implement such a complex thing within `LASlib`. `LASlib` and `LAStools` are now maintained by somebody else, somehow related to Martin's family for what I've heard. The support of copc in `lidR` will come automatically if it is added in `LASlib`. So it is a discussion you should have with https://github.com/LAStools/LAStools . I'd like to work on it but I will waste my time on an unsuccessful project if I try.

**Reply by youngdjn** [reply]
It looks like LAStools (and thus presumably LASlib) now supports COPC: https://github.com/LAStools/LAStools?tab=readme-ov-file#open-source-tools
How would this be accessed via lidR?

**Reply by Jean-Romain** [reply]
`rlas` supports COPC in read not in write yet. `lasR` does. I did the implementation in LASlib a year ago

**Reply by youngdjn** [reply]
Ah that's awesome! I needed a nudge to switch to `lasR` and I think this is it! Thanks for all your work developing these tools.

**Reply by Jean-Romain** [reply]
You don't need to switch to lasR. lasR does not come in replacement of lidR. It is another approach it suits better for practitionner but less for research. Two different approches, two different strategies, two different usages.

**Reply by youngdjn** [reply]
OK I see, thank you. I'm moving toward more practitioner uses anyway. I'll look more closely at the differences.
